ffi/capi/src/neo_datetime.rs

```diff
@@ -0,0 +1,505 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu_datetime::options::FractionalSecondDigits;
+
+#[diplomat::bridge]
+#[diplomat::abi_rename = "icu4x_{0}_mv1"]
+#[diplomat::attr(auto, namespace = "icu4x")]
+pub mod ffi {
+    use alloc::boxed::Box;
+    use diplomat_runtime::DiplomatOption;
+    use icu_datetime::fieldsets::enums::CompositeDateTimeFieldSet;
+    use writeable::Writeable;
+
+    use crate::{date::ffi::IsoDate, datetime_formatter::ffi::DateTimeLength, time::ffi::Time};
+
+    #[cfg(feature = "buffer_provider")]
+    use crate::provider::ffi::DataProvider;
+    #[cfg(any(feature = "compiled_data", feature = "buffer_provider"))]
+    use crate::{
+        errors::ffi::{DateTimeFormatterBuildOrLoadError, DateTimeFormatterLoadError},
+        locale_core::ffi::Locale,
+    };
+
+    #[diplomat::enum_convert(icu_datetime::options::Alignment, needs_wildcard)]
+    #[diplomat::rust_link(icu::datetime::Alignment, Enum)]
+    pub enum DateTimeAlignment {
+        Auto,
+        Column,
+    }
+
+    #[diplomat::enum_convert(icu_datetime::options::YearStyle, needs_wildcard)]
+    #[diplomat::rust_link(icu::datetime::YearStyle, Enum)]
+    pub enum YearStyle {
+        Auto,
+        Full,
+        WithEra,
+    }
+
+    #[diplomat::rust_link(icu::datetime::TimePrecision, Enum)]
+    pub enum TimePrecision {
+        Hour,
+        Minute,
+        MinuteOptional,
+        Second,
+        SecondF1,
+        SecondF2,
+        SecondF3,
+        SecondF4,
+        SecondF5,
+        SecondF6,
+        SecondF7,
+        SecondF8,
+        SecondF9,
+    }
+
+    #[diplomat::enum_convert(icu_datetime::fieldsets::builder::DateFields, needs_wildcard)]
+    #[diplomat::rust_link(icu::datetime::DateFields, Enum)]
+    pub enum DateFields {
+        D,
+        MD,
+        YMD,
+        DE,
+        MDE,
+        YMDE,
+        E,
+        M,
+        YM,
+        Y,
+    }
+
+    #[diplomat::enum_convert(icu_datetime::fieldsets::builder::ZoneStyle, needs_wildcard)]
+    #[diplomat::rust_link(icu::datetime::ZoneStyle, Enum)]
+    pub enum ZoneStyle {
+        Z,
+        Zs,
+        O,
+        Os,
+        V,
+        Vs,
+        L,
+    }
+
+    #[diplomat::rust_link(icu::datetime::builder::FieldSetBuilder, Enum)]
+    pub struct DateTimeFieldSetBuilder {
```

Comment:
is this really a builder? it's useable without having to call `.build()` on it, it's more of an options bag. consider renaming to `DateTimeFieldSet`